packages/nimble-components/src/tooltip/specs/README.md

```diff
@@ -14,6 +14,102 @@ The nimble-tooltip project will first be implemented as a prototype, open issues
+<nimble-button id="previewButton">Preview</nimble-button>
+```
+
+**Pros**
+- No additional implementation cost
+- Familiar pattern, e.g. `<label>` element with `for` attribute, `aria-labelledby` attribute, `aria-describedby` attribute
+- Reasonable DOM structure, i.e. tooltip sibling of anchor element
+
+**Cons**
+- Annoying to have to give anchor element an `id`
+- <span style="color:red">**Not conducive to use in shared components because `id` values must be unique**</span>
+
+### Tooltip slotted in anchor element
+The anchor element defines a `slot` for the tooltip and takes the tooltip as content.
+```html
+<nimble-button>
+    Preview
+    <nimble-tooltip slot="tooltip">...</nimble-tooltip>
+</nimble-button>
+```
+
+**Pros**
+- Familiar slotting pattern
+- Logical DOM structure (element contains its tooltip)
+- May not need `nimble-tooltip` element at all -- could slot tooltip content directly without wrapping in `nimble-tooltip` element
+
+**Cons**
+- Expensive to implement and maintain. <span style="color:red">**Requires forking and modifying the template of every component that wants to support a tooltip.**</span> (Though code duplication could be avoided by using shared template fragments.) Tooltip implementation leaks into every other component.
+- Cannot apply tooltip to native elements, like `<div>`
+
+### Tooltip implemented by each component
+There is no tooltip component, but instead, each component that wants to support a tooltip makes it a part of its template. Those components have an attribute like `tooltip` that takes the text to display in the tooltip. If the tooltip needs to support more than just unformated text, that approach would be a variation of "Tooltip slotted in anchor element", where there is no explicit tooltip element.
+```html
+<nimble-button tooltip="This is tooltip text">Preview</nimble-button>
+```
+
+**Pros**
+- Simple, intuitive API (similar to existing `title` attribute)
+- No dedicated tooltip component
+
+**Cons**
+- Expensive to implement and maintain. <span style="color:red">**Requires forking and modifying the template of every component that wants to support a tooltip.**</span> (Though code duplication could be avoided by using shared template fragments.) Cannot leverage FAST implementation?
+- <span style="color:red">**Does not support rich tooltip content (which is probably a requirement)**</span>
+- Cannot apply tooltip to native elements, like `<div>`
+
+### Tooltip takes anchor element as content
+The tooltip element wraps the anchor element. If the tooltip takes other content (e.g. the text for the tooltip itself), the anchor element would need to specify something like `slot="anchor"` to distinguish it from that other content.
+```html
+<nimble-tooltip>
+    <nimble-button slot="anchor">Preview</nimble-button>
+    This is the tooltip text.
+</nimble-tooltip>
+```
+
+**Pros**
+- Easy to implement
+- Proven pattern: used by Carbon and Atlassian Design Systems
```

Comment:
I think "proven" is a strong assertion, I'd be interested in knowing if those design system authors and their users actually prefer that pattern and if so how they came to that design.

```suggestion
- Pattern used by Carbon and Atlassian Design Systems
```